perception/autoware_multi_object_tracker/src/multi_object_tracker_node.cpp

```diff
@@ -188,19 +188,34 @@ MultiObjectTracker::MultiObjectTracker(const rclcpp::NodeOptions & node_options)
 
     AssociatorConfig associator_config;
     {
-      const auto tmp = this->declare_parameter<std::vector<int64_t>>("can_assign_matrix");
-      const std::vector<int> can_assign_matrix(tmp.begin(), tmp.end());
-      associator_config.can_assign_matrix = can_assign_matrix;
+      auto initializeMatrixInt = [](const std::vector<int64_t> & vector) {
+        const int label_num = static_cast<int>(std::sqrt(vector.size()));
+        std::vector<int> converted_vector(vector.begin(), vector.end());
+        Eigen::Map<Eigen::MatrixXi> matrix_tmp(converted_vector.data(), label_num, label_num);
+        // transpose to make it row-major
+        return matrix_tmp.transpose();
+      };
+      auto initializeMatrixDouble = [](const std::vector<double> & vector) {
+        const int label_num = static_cast<int>(std::sqrt(vector.size()));
+        Eigen::Map<Eigen::MatrixXd> matrix_tmp(
+          const_cast<double *>(vector.data()), label_num, label_num);
```

Comment:
How about just return const and not using const_cast? Because input is const and we do not intend to modify input in this conversion.

```suggestion
        Eigen::Map<const Eigen::MatrixXd> matrix_map( 
            vector.data(), label_num, label_num
        );
```

Reply:
fixed https://github.com/autowarefoundation/autoware_universe/pull/10456/commits/aa85148a584392de201db453957bd36ee39eb2ea